### PR TITLE
[MM-24463] Run message retention cleanup off of pre-existing state instead of a reconstructed "zero" state (1.30)

### PR DESCRIPTION
**Pull request by chuttam**
(**This is the PR for 1.30. Separate PR to follow for `master` and 1.31**)

#### Summary

Only posts in channels, searched posts and flag posts are recalculated (as per data retention policy, if applicable). The rest of state is cloned from existing state.

On a version upgrade, data cleanup gets run anyway.

#### Ticket Link

* [MM-24463](https://mattermost.atlassian.net/browse/MM-24463)

#### Device Information
This PR was tested on: OnePlus 5 (Android 10 device)

### CODE REVIEW

**Comment by chuttam** [comment]
@migbot This alters [your recent work][https://github.com/mattermost/mattermost-mobile/pull/4197/files#diff-2710a8d89549f3295d6a4f1d1c6cca7dR30-R49] a little bit, and I think I've retained the desired functionality. But pointing out here just to draw a bit of extra attention.

**Comment by chuttam** [comment]
Added @ogi-m for QA Review only because of previous, similar testing context done for ticket [MM-24426]( https://mattermost.atlassian.net/browse/MM-24426). Please change if not appropriate.

File: app/store/middleware.js [hunk 1]
```diff
@@ -35,11 +33,6 @@ export function messageRetention(store) {
                 return next(action);
             }
```
Comment (enahum) [inline app/store/middleware.js:35]: 🏖

**Comment by enahum** [comment]
@saturninoabril, what do you mean by “and 1.30” in your first bullet item?

**Comment by saturninoabril** [comment]
Oh, please disregard, I'm totally off there as I thought we're currently working for v1.31. It's v1.30 build 285 where I tested upgrade into this PR build.

**Comment by chuttam** [comment]
> No changes -- Patch already applied.

Changes were merged to `release-1.30` branch.

**Comment by saturninoabril** [comment]
I didn't got a chance to test something related to data retention policy as requested by EN due to some issues I faced when setting up the server, but will find time to do so separately.

**Pull request merged**